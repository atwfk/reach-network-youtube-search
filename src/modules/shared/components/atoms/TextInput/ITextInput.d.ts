declare namespace ITextInput {
  export interface IProps {
    name: string;
    id: string;
    placeholder: string;
    value: string;
    changed: (value: string) => void;
    deleteText: () => void;
  }
}

export { ITextInput };
