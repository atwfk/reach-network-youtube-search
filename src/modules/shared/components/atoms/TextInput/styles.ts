import styled from "styled-components";
import { SCREENS } from "../../../constants";

export const StyledInputContainer = styled.div`
  position: relative;
  display: flex;
  align-items: center;
  width: 100%;

  input {
    width: 100%;
    padding: 5px;
    border: 1px solid #9d9d9d;
    font-size: 16px;
    outline: none;
    border-radius: 2px;
    :focus {
      border-color: #3ea6ff;
    }

    @media (min-width: ${SCREENS.MD}) {
      padding: 10px;
    }
  }

  button {
    position: absolute;
    right: 10px;
    top: 50%;
    transform: translateY(-50%);
    display: flex;
    cursor: pointer;

    @media (min-width: ${SCREENS.MD}) {
      right: 20px;
    }
  }
  .remove-text {
    color: #9d9d9d;
  }
`;
