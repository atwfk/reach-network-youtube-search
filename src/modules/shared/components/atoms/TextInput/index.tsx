import React from "react";
import type { FC, ReactElement } from "react";
import { AiOutlineClose } from "react-icons/ai";
import type { ITextInput } from "./ITextInput";
import { StyledInputContainer } from "./styles";

const TextInput: FC<ITextInput.IProps> = ({
  name,
  id,
  placeholder,
  value,
  changed,
  deleteText
}): ReactElement => {
  return (
    <StyledInputContainer>
      <input
        type="text"
        data-testid="text-input"
        name={name}
        id={id}
        placeholder={placeholder}
        value={value}
        onChange={(e) => changed(e.currentTarget.value)}
      />
      {value && (
        <button type="button" onClick={deleteText} data-testid="delete-btn">
          <AiOutlineClose className="remove-text" />
        </button>
      )}
    </StyledInputContainer>
  );
};

export default TextInput;
