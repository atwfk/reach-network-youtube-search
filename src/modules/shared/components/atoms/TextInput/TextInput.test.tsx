import { render, screen, cleanup } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import "jest-styled-components";
import renderer from "react-test-renderer";
import TextInput from ".";

afterEach(cleanup);

describe("TextInput Component snapshots", () => {
  const changed = jest.fn();
  const deleteText = jest.fn();

  it("Should render input element without delete icon when input is empty", () => {
    const tree = renderer
      .create(
        <TextInput
          name="search"
          id="1"
          placeholder="Search"
          value=""
          changed={changed}
          deleteText={deleteText}
        />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("Should render input element with delete icon when input has value", () => {
    const tree = renderer
      .create(
        <TextInput
          name="search"
          id="1"
          placeholder="Search"
          value="sabba7"
          changed={changed}
          deleteText={deleteText}
        />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});

describe("TextInput Component RTL", () => {
  const changed = jest.fn();
  const deleteText = jest.fn();

  it("should not have delete icon in the DOM when input is empty", () => {
    render(
      <TextInput
        name="search"
        id="1"
        placeholder="Search"
        value=""
        changed={changed}
        deleteText={deleteText}
      />
    );

    const deleteBtn = screen.queryByTestId("delete-btn");

    expect(deleteBtn).not.toBeInTheDocument();
  });

  it("should have delete icon in the DOM when input has value", () => {
    render(
      <TextInput
        name="search"
        id="1"
        placeholder="Search"
        value="sabba7"
        changed={changed}
        deleteText={deleteText}
      />
    );

    const deleteBtn = screen.getByTestId("delete-btn");

    expect(deleteBtn).toBeInTheDocument();
  });

  it("should call onChange when we start typing in the input", () => {
    render(
      <TextInput
        name="search"
        id="1"
        placeholder="Search"
        value=""
        changed={changed}
        deleteText={deleteText}
      />
    );

    const input = screen.getByTestId("text-input");
    userEvent.type(input, "aa");

    expect(changed).toHaveBeenCalledTimes(2);
  });

  it("should call deleteText when we click on delete icon", () => {
    render(
      <TextInput
        name="search"
        id="1"
        placeholder="Search"
        value="aa"
        changed={changed}
        deleteText={deleteText}
      />
    );

    const deleteBtn = screen.getByTestId("delete-btn");

    userEvent.click(deleteBtn);

    expect(deleteText).toHaveBeenCalled();
  });
});
