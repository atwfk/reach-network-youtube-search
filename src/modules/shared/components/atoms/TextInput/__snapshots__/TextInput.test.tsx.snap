// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput Component snapshots Should render input element with delete icon when input has value 1`] = `
.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
}

.c0 input {
  width: 100%;
  padding: 5px;
  border: 1px solid #9d9d9d;
  font-size: 16px;
  outline: none;
  border-radius: 2px;
}

.c0 input:focus {
  border-color: #3ea6ff;
}

.c0 button {
  position: absolute;
  right: 10px;
  top: 50%;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  cursor: pointer;
}

.c0 .remove-text {
  color: #9d9d9d;
}

@media (min-width:768px) {
  .c0 input {
    padding: 10px;
  }
}

@media (min-width:768px) {
  .c0 button {
    right: 20px;
  }
}

<div
  className="c0"
>
  <input
    data-testid="text-input"
    id="1"
    name="search"
    onChange={[Function]}
    placeholder="Search"
    type="text"
    value="sabba7"
  />
  <button
    data-testid="delete-btn"
    onClick={[MockFunction]}
    type="button"
  >
    <svg
      className="remove-text"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 1024 1024"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M563.8 512l262.5-312.9c4.4-5.2.7-13.1-6.1-13.1h-79.8c-4.7 0-9.2 2.1-12.3 5.7L511.6 449.8 295.1 191.7c-3-3.6-7.5-5.7-12.3-5.7H203c-6.8 0-10.5 7.9-6.1 13.1L459.4 512 196.9 824.9A7.95 7.95 0 0 0 203 838h79.8c4.7 0 9.2-2.1 12.3-5.7l216.5-258.1 216.5 258.1c3 3.6 7.5 5.7 12.3 5.7h79.8c6.8 0 10.5-7.9 6.1-13.1L563.8 512z"
      />
    </svg>
  </button>
</div>
`;

exports[`TextInput Component snapshots Should render input element without delete icon when input is empty 1`] = `
.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
}

.c0 input {
  width: 100%;
  padding: 5px;
  border: 1px solid #9d9d9d;
  font-size: 16px;
  outline: none;
  border-radius: 2px;
}

.c0 input:focus {
  border-color: #3ea6ff;
}

.c0 button {
  position: absolute;
  right: 10px;
  top: 50%;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  cursor: pointer;
}

.c0 .remove-text {
  color: #9d9d9d;
}

@media (min-width:768px) {
  .c0 input {
    padding: 10px;
  }
}

@media (min-width:768px) {
  .c0 button {
    right: 20px;
  }
}

<div
  className="c0"
>
  <input
    data-testid="text-input"
    id="1"
    name="search"
    onChange={[Function]}
    placeholder="Search"
    type="text"
    value=""
  />
  
</div>
`;
